lib/jekyll.rb

```diff
@@ -49,12 +49,19 @@ module Jekyll
   autoload :Filters,             'jekyll/filters'
   autoload :FrontmatterDefaults, 'jekyll/frontmatter_defaults'
   autoload :Layout,              'jekyll/layout'
-  autoload :LayoutReader,        'jekyll/layout_reader'
+  autoload :CollectionReader,    'jekyll/readers/dynamic/collection_reader'
+  autoload :DataReader,          'jekyll/readers/dynamic/data_reader'
+  autoload :LayoutReader,        'jekyll/readers/dynamic/layout_reader'
+  autoload :DraftReader,         'jekyll/readers/dynamic/draft_reader'
+  autoload :PostReader,          'jekyll/readers/dynamic/post_reader'
+  autoload :PageReader,          'jekyll/readers/dynamic/page_reader'
+  autoload :StaticFileReader,    'jekyll/readers/static/static_file_reader'
```

Comment:
Good point. I thought there were going to be more static readers. Therefore initially I split it up in the two directories. But yes you're right it's a bit overdone like this. I'll change it now.